tools/clang/lib/Parse/ParseStmt.cpp

```diff
@@ -1582,7 +1582,7 @@ StmtResult Parser::ParseForStatement(SourceLocation *TrailingElseLoc) {
     ScopeFlags = Scope::DeclScope | Scope::ControlScope;
 
   // HLSL Change Starts - leak declarations in for control parts into outer scope
-  if (getLangOpts().HLSL) {
+  if (getLangOpts().HLSLVersion < 2020) {
```

Comment:
It makes little difference to the end result, but referencing non-existent version 2020 makes me just a bit uncomfortable. The way these kind of checks are done elsewhere is of the form `HLSLVersion < 2021`.

The only reason I can think of is if someone wants to search for all the places that HLSLVersion 2021 changes the behavior and would miss this.

Reply:
My brain is still slightly stuck in March 2020... I think today's date is 3/626/2020...